spec/dictionary_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'dictionary'
+
+RSpec.describe Dictionary do
+  it 'is initial empty' do
+    dictionary = Dictionary.new
+
+    expect(dictionary).to be_empty
+  end
+
+  it 'is not empty after adding word and definition' do
+    dictionary = Dictionary.new
+
+    dictionary.add('Dog', [])
+
+    expect(dictionary).not_to be_empty
+  end
+
+  it 'deletes given word and the related definitions' do
+    dictionary = Dictionary.new
+
+    dictionary.add('Dog', ['Definition 1'])
+    dictionary.delete_word('Dog')
+
+    expect(dictionary.definition('Dog')).to eq(nil)
+  end
+
+  it 'deletes a definition of a given word' do
+    dictionary = Dictionary.new
+
+    dictionary.add('Dog', ['Definition 1', 'Definition 2'])
+    dictionary.delete_definition('Dog', 'Definition 2')
+
+    expect(dictionary.definition('Dog')).to eq(['Definition 1'])
+  end
+
+  it 'gets the definitions of a word' do
+    dictionary = Dictionary.new
+
+    dictionary.add('Dog', ['Definition 1', 'Definition 2'])
+
+    expect(dictionary.definition('Dog')).to eq(['Definition 1', 'Definition 2'])
+  end
+
+  it 'appends a new definition to existing entry' do
+    dictionary = Dictionary.new
+
+    dictionary.add('Dog', ['Definition 1', 'Definition 2'])
+    dictionary.append('Dog', ['Definition 3'])
```

Comment:
```suggestion
    dictionary.append('Dog', 'Definition 3')
```
I think this interface makes much more sense.
In case you want to append multiple definitions at once, I suggest another method or append is able to handle both, string and array of string.